pkg/controller/jobframework/setup.go

```diff
@@ -33,6 +36,7 @@ import (
 
 var (
 	errFailedMappingResource = errors.New("restMapper failed mapping resource")
+	restMapperMutex          sync.RWMutex
```

Comment:
To avoid confusion other members, please, add a comment why we add this mutex (that it is using only for unit test and we need to lock RESTMapper on update).